pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -368,10 +389,14 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 
 		for i := 0; i < pages; i++ {
 			results := searchData.GetSearch()
-			searchRepos = append(searchRepos, results.Nodes...)
+			for _, repo := range results.Nodes {
+				if r, ok := repo.(*SearchNodeRepository); ok {
+					searchRepos = append(searchRepos, *r)
+				}
+			}
 
 			repoCursor = &searchData.Search.PageInfo.EndCursor
-			data, err = getRepoData(ctx, genClient, sq, ownertype, repoCursor)
+			searchData, err = getRepoData(ctx, genClient, sq, ownertype, repoCursor)
```

Comment:
Yeah this section is weird I just hadn't bothered seeing how to change it since that's how it was when i started working on it.  The first bit of this data we grab is outside the for loop and we use that to get the amount of pages we need to iterate through but we also still just have the first page of data, so it gets processed first, then we grab the next page to be processed on the next loop at the end.  On the other functions like this, we have a graphql query to get the count first to avoid this so that might be an option here as well, just havnt checked it out yet.